DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -396,7 +396,16 @@ Function Upgrade-Package {
 }
 
 function Get-ChocoInstalledPackage {
-    Return (choco list -lo -r | ConvertFrom-Csv -Header 'Name', 'Version' -Delimiter "|")
+    $PackageCacheSec = (Get-Date).AddSeconds('-60')
+    $ChocoInstallList = Join-Path -Path $env:ChocolateyInstall -ChildPath 'ChocoInstalled.xml'

```

Comment:
can we put this to a cache folder under the install? Something like `$env:ChocolateyInstall\cache\list`?

Reply:
Ok I looked into it, because its not a default folder path created by a choco install and `Export-Clixml` does not support creating a directory structure. We have to `Test-Path` to *\cache and create if missing on every `Get-ChocoInstalledPackage` run.  This actually only adds about 2 seconds on 10,000 runs (on my ssd) so probably not a big deal for the sake of keeping things clean.